src/Microsoft.NetCore.Analyzers/Core/Performance/UsePropertyInsteadOfCountMethodWhenAvailable.cs

```diff
@@ -234,7 +234,7 @@ internal bool IsICollectionOfTImplementation(ITypeSymbol invocationTarget)
                 return false;
 
                 bool isCollectionOfTInterface(ITypeSymbol type)
-                    => this.ICollectionOfTType.Equals(type.OriginalDefinition);
+                    => this.ICollectionOfTType!.Equals(type.OriginalDefinition);
```

Comment:
💡 If you make this a static local function, the suppression would not be required.